book/python/using-connectors.md

```diff
@@ -0,0 +1,193 @@
+
+Care has been taken to keep the API simple and the runtime assumtions unobtrusive so it's easy to hook up the library to existing code. Read down below for more information on how this works.
+
+Connectors themselves are composed of to two parts. First is the entry in your application builder which explains how encoding and decoding of the content works as well as giving each connector instance a name and direction of flow (source or sink). Second is the script which is a local process which runs side-by-side with Wallaroo. We'll look at each part in turn in the sections below.
+
+## Using A Connector In Your Application
+
+In your application module's `application_setup` function, the application builder returns a description of your application in terms of pipelines. Connectors tap into this by allowing the declaration of source and sink connectors and wiring these up by name in your pipeline defitions.
+
+Here is an example of what this looks like:
+
+```python
+def application_setup(args):
+    ab = wallaroo.ApplicationBuilder("Celsius to Fahrenheit")
+
+    ab.source_connector("celsius_feed",
+        encoder=encode_feed,
+        decoder=decode_feed)
+
+    ab.sink_connector("fahrenheit_conversion",
+        encoder=encode_conversion,
+        decoder=decode_conversion)
+
+    ab.new_pipeline("Celsius Conversion", "celsius_feed")
+    ab.to(multiply)
+    ab.to(add)
+    ab.to_sink("fahrenheit_conversion")
+    return ab.build()
+```
+
+You can follow along in this example by going to [github](https://github.com/Wallaroo/wallaroo/tree/{{book.wallaroo_version}}/examples/python/celsius_connectors/).
+
+We have introduced two new application builder methods for declaring connectors: `source_connector` and `sink_connector`. These allow you to describe both ends of the connection so the connector script can encode or decode data in a way that's compatible with your application. Keeping these in one place helps ensure that it's easy to keep these in sync.
+
+In our example we can look at the celsius feed's encoder and decoder functions:
+
+```python
+@wallaroo.experimental.stream_message_encoder
+def encode_feed(data):
+    return data
+
+@wallaroo.experimental.stream_message_decoder
+def decode_feed(data):
+    return struct.unpack(">f", data)[0]
+```
+
+Here we've defined two functions and use `wallaroo.experimental.stream_message_encoder` and `wallaroo.experimental.stream_message_decoder` as decorators. These decorators will handle framing and delivery of your data, so all you need to ensure is that the bytes represent the single message you want to deliver.
+
+A good observer will note that we're just returning the data in this example's `encode_feed` function. In this case we're receiving data that's already a packed floating point number so there is no need to reencode it. This allows you to pass along data as-is when it's already in the correct format.
```

Comment:
Can we ditch "a good observer". It can be judgemental for someone who perhaps didn't note that.

Reply:
How's "You might want to ask why we're just returning .."?

Reply:
"You might ask why we're returning the data unchanged... " ?

I think something like that is a little more clear then "just returning"